Issue160 Use cucumber.js instead of behave

raar1 opened the pull request:
Fixes PerfectFit-project/virtual-coach-issues#160.

svenvanderburg commented on script/bootstrap:
```diff
@@ -8,11 +8,10 @@ else git clone https://github.com/PerfectFit-project/niceday-components.git buil
 
 
-cd build/niceday-components/niceday-broker/ || exit 1
+cd niceday-broker/ || exit 1
```
Ah, this is what Candace also ran into I guess?

svenvanderburg replied:
And #13 fixes this in a nicer way?

raar1 replied:
No, I think it's the other way around. This is showing up because of a merge conflict - Candace's fix is actually newer and works better. I'll fix this.

svenvanderburg commented on README.md:
```diff
@@ -37,11 +37,11 @@ Run `script/server` script to serve the application, or follow these steps:
 
 ## Test
-Run `script/test` script to serve the application and run `behave`, or follow these steps:
+Run `script/test` script to serve the application and run `script/cucumber`, or follow these steps:
```
The cucumber tests should actually run right when we just run `script/test`?

raar1 replied:
Yes, good catch. I've fixed this in the README

svenvanderburg reviewed (approved):
Nice :) I added a few comments in the code.
My main comment is this: we still test on the level of rasa right? It could be that there is an error in the broker because we don't test that. 

What we should do instead of doing a request to rasa, we use `goalie-js` to send a message from the user to the therapist account. This should be picked up by the broker, which then sends it to rasa. The response from rasa is send back to the user. In our test we listen for incoming messages from the therapist account to the user. Then we check if the expected message is returned. Then we have a full end-to-end test. 

This will include some more annoying dependencies (i.e. goalie-js) for running the e2e-test, but I think it is worth it. What do you think? If you like it please create an issue for it. I am also curious to hear if you can think of something more clever.

raar1 commented:
Thanks for the review, Sven. I've addressed the comments, and made a new issue for the (more comprehensive) e2e testing: https://github.com/PerfectFit-project/virtual-coach-issues/issues/193
